packages/ord-connect/src/components/SelectWalletModal/index.tsx

```diff
@@ -72,6 +75,76 @@ export function SelectWalletModal({
     [disconnectWallet],
   );
 
+  const onConnectMagicEdenWallet = useCallback(async () => {
+    if (network === "testnet") {
+      const unsupportedNetworkError = new Error(
+        "Magic Eden wallet is not supported on testnet",
+      );
+      onError(Wallet.MAGICEDEN, unsupportedNetworkError);
+      return false;
+    }
+
+    try {
+      setErrorMessage("");
+      const magicEden = await getMagicEdenAddress(network);
+      if (!magicEden || magicEden.length < 1) {
+        disconnectWallet();
+        throw new Error("Magic Eden via Ordit returned no addresses.");
+      }
+
+      // Magic Eden provides segwit address for sending and receiving payments
+      const segwit = magicEden.find(
+        (walletAddress) => walletAddress.format === "segwit",
+      );
+      // p2sh is used to support imported xverse wallets
+      const p2sh = magicEden.find(
+        (walletAddress) => walletAddress.format === "p2sh-p2wpkh",
+      );
```

Comment:
can we consolidate these find calls? if one of them is only present, the first will be selected.

```suggestion
      // segwit used for payments in Magic Eden, imported wallets may use p2sh-p2wpkh
      const paymentAddress = magicEden.find(
        (walletAddress) => walletAddress.format === "segwit" || walletAddress.format === "p2sh-p2wpkh",
      );
```

Reply:
Fixed